components/contracts/contractRouting.ts

```diff
@@ -329,12 +329,14 @@ function getContractOpportunityData(
         false,
     )
 
-    for (const ms of contract.Metadata.Opportunities) {
-        missionStories[ms].PreviouslyCompleted =
-            ms in userData.Extensions.opportunityprogression
-        const current = fastClone(missionStories[ms])
-        delete current.Location
-        result.push(current)
+    if (contract.Metadata.Opportunities) {
```

Comment:
I think it would be better to have this as `if (!contract.Metadata.Opportunities) return result;` but that's just me.